<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons+Sharp" rel="stylesheet">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@NaN,0,0,0" />
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200" />
    <link rel="stylesheet" href="../static/dashboard.css">
    <title>Dashboard | EVOLVE</title>
</head>

<body>

    <div class="container">
        <!-- Sidebar Section -->
        <aside>
            <div class="toggle">
                <div class="logo">
                    <img src="../static/images/logo.png">
                    <h2>EVO<span class="danger">LVE</span></h2>
                </div>
                <div class="close" id="close-btn">
                    <span class="material-icons-sharp">
                        close
                    </span>
                </div>
            </div>

            <div class="sidebar">
                <a href="/dashboard">
                    <span class="material-icons-sharp">dashboard</span>
                    <h3>Dashboard</h3>
                </a>
                <a href="/preprocessing">
                    <span class="material-symbols-outlined">schema</span>
                    <h3>Preprocessing</h3>
                </a>
                <a href="/entropy">
                    <span class="material-icons-sharp">bar_chart</span>
                    <h3>Entropy Calculation</h3>
                </a>
                <a href="/mrf">
                    <span class="material-symbols-outlined">schema</span>
                    <h3>Mutational Response Function</h3>
                </a>
                <a href="/feature1">
                    <span class="material-symbols-outlined">stat_1</span>
                    <h3>Feature 1</h3>
                </a>
                <a href="/feature3">
                    <span class="material-symbols-outlined">stat_3</span>
                    <h3>Feature 3</h3>
                </a>
                <a href="/prediction">
                    <span class="material-symbols-outlined">batch_prediction</span>
                    <h3>Prediction</h3>
                </a>
                <a href="/tutorial_dashboard" class="active">
                    <span class="material-symbols-outlined">psychology</span>
                    <h3>Tutorial</h3>
                </a>
                <a href="/logout">
                    <span class="material-icons-sharp">logout</span>
                    <h3>Logout</h3>
                </a>
            </div>
        </aside>
        <!-- End of Sidebar Section -->

        <!-- Main Content -->
        <main>
            <h1>Tutorial</h1>
            <!-- New Users Section -->
            <div class="new-users">
                <div class="user-list">
                    <h2>Step 1: Downloading the Dataset</h2>
                    <p>In this step, you will need to download a dataset of protein/genomic sequences either from the National
                        Center for Biotechnology Information, Virus Database NCBI Virus or the Global Initiative on Sharing Avian
                        Influenza Data GISAID. NCBI and GISAID are two major databases for accessing genetic sequence data, and you
                        can choose either one based on the specific research question you are trying to answer.</p>
                    <p>To download the dataset, you will need to navigate to the <a href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/" target="_blank">NCBI Virus </a>
                        or <a href="https://gisaid.org/" target="_blank">GISAID </a>website and search for the
                        dataset you want to download. You can filter the search results based on the specific criteria you are
                        interested in, such as the type of virus or the geographic region. Once you have found the dataset you want
                        to download, you can download it in a compressed file format, such as .fasta or .fastq</p>
                    <p>In our previous <a href="https://pubs.acs.org/doi/10.1021/acs.jpcb.2c04574" target="_blank">work </a>we
                        downloaded the protein data from NCBI Virus, in a month-wise manner, by selecting certain filters. Follow
                        the following steps to download the data in a month-wise manner</p>
                    <h3><i>NCBI Virus Database to Download Protein Sequence</i></h3>
                    <p>Go to NCBI Virus dashboard. Click on the Country for which you want to download the data. In this case we select India and
                        then scroll up and select All Proteins</p>
                    <p>Here you will find the details of all the viral proteins in that country. From the left panel
                        select filters to narrow down the search to a specific variant. Click on Nucleotide Completeness and
                        select Complete. This step makes sure that you are using a complete sequence and not a partial or
                        incomplete sequence</p>
                    <p>Once you have selected the complete sequence, click on Pango Lineage to select the variant of
                        interest. In this case we are selecting the Delta Variant (B.1.617.2). Make sure that you enter the
                        variant ID and not the generic name of the variant.</p>
                    <p>Once you have selected the variant of your choice, then click on Proteins. Here you can select
                        which viral protein you would like to investigate. In this case we have selected the Surface
                        Glycoprotein of Delta Variant</p>
                    <p>Click on Collection Date to display the sequences in a month wise manner. Finally to download the
                        sequences click on Download and proceed with the default options</p>
                    <h3><i>GISAID Database to Download Genomic Sequence</i></h3>
                    <p>You need to make an account in GISAID first. Go to GISAID and in the top right corner click on Log
                        In. Create an account and once you log in you will see the home page. Make sure you are on the EpiCov tab. Click on the Viral structure of SARS-CoV-2 and it will take
                        you to the working dashboard</p>
                    <p>Once you are on the working dashboard navigate to Location tab. Here you can search the genomic
                        sequence based on the countries. But there is a format that you need to use to search. First type the
                        name of the Continent followed by a space and then a Forward Slash, followed by the name of the country.
                        For example if we need to search sequences from India then we will type Asia / India</p>
                    <p>After searching for the country of choice, navigate to the right side to select three important
                        features: Complete, High Coverage and Collection Date Complete. Selecting Complete assures that the
                        sequences that you will be getting are not incomplete or broken. Selecting High Coverage affirms that
                        the genomic sequences that will be displayed has less amount of unrecognised base pairs i.e. low value
                        of N's (<1% N and <0.05% mutations). And finally selecting Collection Date Complete allows the server
                        to give you entries with complete collection date.</p>
                    <p>Once the filters are selected then you can navigate to Lineage tab to select the variant of your
                        choice. For example if we want to download the genomic sequence of Delta variant then in the Lineage
                        tab we will type the ID of the Delta variant (i.e. B.1.617.2).</p>
                    <p>Once you have selected the variant of interest then you can navigate to the Collection Date and
                        collect the sequences in a month wise manner</p>
                    <p>After performing the above steps you will get the refined sequences in a month-wise manner. To
                        download the sequences you can click on the Square Box on the left side of Virus Name tab. This will
                        select all the sequences in the entry and then you can navigate to the bottom right and click on the
                        Download button to download the dataset. Once you click on Download a new window will pop up asking
                        you to select the format of the dataset. Click on Nucleotide Sequences (FASTA) to download the dataset
                        in FASTA format and then click on Download. Then you will be asked to accept the terms and conditions
                        of GISAID database. Click on I agree to the terms and conditions and then click Download.</p>
                    <h2>Step 2: Dataset Preprocessing</h2>
                    <p>After downloading the dataset it is important to preprocess it as the raw data consists of many junk and
                        broken data.To preprocess the data click on Preprocessing tab from the main menu and then upload the sequence
                        file that you have downloaded in previous tep in FASTA format. Click on Submit after which you will get your
                        preprocessed data. To know more about what exactly the server does in preprocessing please visit the Theory
                        Corner.</p>
                    <p>If you are unsure about which data file or what type of datafile you need to upload, you click on Load
                        Example and then click on Submit. Once the preprocessing is done, you can download the file and then view
                        it to understand the file type and its architecture.</p>
                    <h2>Step 3: Entropy Calculation</h2>
                    <p>After you have preprocessed the data the first thing that we need to do is to calculate the entropy.
                        Before you can calculate the Mutational entropy, you need to have the alignment file with you for the
                        preprocessed data. You can visit <a href="https://mafft.cbrc.jp/alignment/server/" target="_blank">MAFFT Server</a>
                        to perform the Multiple Sequence Alignment and Download the alignment file.</p>
                    <p>On the MAFFT Server, upload your preprocessed FASTA file by clicking on "Choose File" and then click on "Submit".
                        You can either keep the deafult values as it is or can change it according to your own need. Once the alignment is
                        done, you can download the alignment file (.aln) from the top left corner by clicking on "Clustal Format".</p>
                    <p>Upload the clustal alignment file (.aln) in our webserver and click on "Submit". You will get the desired output
                        in the form of Mutational Entropy graph</p>
                    <p>If you are unsure about which data file or what type of datafile you need to upload, you can click on Load
                        Example and then click on Submit. Once the preprocessing is done, you can download the file and then view
                        it to understand the file type and its architecture. In the output you will get the graph of Entropy vs Residue number. This graph depicts the Mutational Entropy
                        of each residue in the viral strain</p>
                    <h2>Step 4: Mutational Response Function</h2>
                    <p>After calculating the Entropy of the genomic/protein sequence you need to supply the data file in order
                        to calculate the MRF. When you download the entropy data in the previous step you will get a CSV file. If you look at the
                        architecture of the CSV file you will see that the first column of the CSV file is the "Residue Position" and the
                        second column is the "Entropy". You need to prepare the correct data in order to calculate the MRF as well</p>
                    <p>The CSV file should have the following architecture: The first column will correspond to "Residue Position",
                        The second column will correspond to "Entropy" of a particular month. Depending upon the average month that
                        you are going to take (default is 3), you need to increase or decrease the number of months. So if you want
                        to calculate the MRF for 5 months, for example, then the first column will be "Residue Position" and then second,
                        third, fourth, fifth and sixth columns will correspond to the entropy of the individual months. You can
                        download the example file to see the architecture of the file.</p>
                    <h2>Step 5: Feature Calculation</h2>
                    <p>This step is totally different from the previous steps as in this step you will be calculating the features
                        necessary to train the model. There are four major features that you are required to calculate in order to
                        train the model</p>
                    <h3><i>Feature 1: Pair Predictability of Amino Acids</i></h3>
                    <p>To calculate this feature click on "Feature 1" on the left panel in your dashboard. Upload the
                        preprocessed fasta sequence file and click on "Submit" and this will calculate the Feature 1 of all the
                        sequences present in your fasta file</p>
                    <p>In the output you will get a zip file to download. Download the zip file and extract it. The folder will consists of
                        CSV files named based on the sequences's Accession ID</p>
                    <p>The architecture of the CSV file will correspond to Residue Position in first column and Feature 1 for each amino acid
                        in second column. To know more about the mathematical background of the Feature 1, please go through
                        the Theory Corner</p>
                    <h3><i>Feature 2: Future Composition of Amino Acids</i></h3>
                    <p>To calculate this feature click on "Feature 2" on the left panel in your dashboard. Upload the
                        preprocessed fasta sequence file and click on "Submit" and this will calculate the Feature 2 of all the
                        sequences present in your fasta file</p>
                    <p>In the output you will get a zip file to download. Download the zip file and extract it. The folder will consists of
                        CSV files named based on the sequences's Accession ID. The architecture of the CSV file will correspond
                        to Residue Position in first column and Feature 2 for each amino acid
                        in second column. To know more about the mathematical background of the Feature 2, please go through the
                        Theory Corner</p>
                    <h3><i>Feature 3: Entropy Calculation of Amino Acids</i></h3>
                    <p>To calculate this feature click on "Entropy Calculation" on the left panel in your dashboard. Upload the
                        preprocessed fasta sequence file and click on "Submit" and this will calculate the Feature 3 of all the
                        sequences present in your fasta file</p>
                    <p>In the output you will get a zip file to download. Download the zip file and extract it. The folder will consists of
                        CSV files named based on the sequences's Accession ID. The architecture of the CSV file will correspond
                        to Residue Position in first column and Feature 3 for each amino acid
                        in second column. To know more about the mathematical background of the Feature 3, please go through
                        the Theory Corner</p>
                    <h2>Phylogenetic Calculation</h2>
                    <p>Once all the features are calculated, you need to perform a phylogenetic analysis in order to get the
                        target class labels. To perform the phylogenetic analysis, you need to first get the phylogenetic tree.
                        To get the phylogenetic tree, upload the preprocessed fasta sequence file in <a href="https://mafft.cbrc.jp/alignment/server/" target="_blank">MAFFT Server</a>
                        and then click on "Submit"</p>
                    <p>Once you get the output page, click on "Phylogenetic Tree" option. Once you click on "Phylogenetic Tree" option a new page will open up. Keep the parameters at their default values
                        unless you want to change anything. Once you have decided on the parameters then click on "Go!"</p>
                    <p>Once you click on "Go!" a new window will come up. Click on "View Tree on Phylo.io" to view the phylogenetic tree. Then
                        visit the <a href="https://github.com/psychedelic2007/biohub_webserver/tree/main/local_calculation" target="_blank">Local Calculation</a>
                        directory of the official GitHub page of the webserver</p>
                    <p>Download the "phylogenetic_calculation.py" python file in your local computer and run the code. When prompted to enter
                        the first sequence, then type the Accession ID of the root sequence (parent) and press enter.
                        When prompted to enter the second sequence, then type the Accession ID of the branch sequence (daughter) and press enter.
                        Make sure to change the "fasta_file_path" to a folder where you would like to save the files. To know more
                        about how to calculate the root sequence (parent) and branch sequence (daughter) please visit the Theory corner</p>
                    <p>Once you have calculated all the necessary features and performed the phylogenetic analysis again visit the
                        <a href="" target="_blank">Local Calculation</a> directory of the official GitHub page of the webserver and download the
                        python files starting with "combine" and then run them in your local computer to generate the combined files which will
                        be the training dataset for the model. To know more about how to run the python codes, visit the
                        "README.md" file in the "Local Calculation" directory</p>
                    <h2>Step 6: Mutation Prediction</h2>
                    <p>After calculating the features for the preprocessed sequences you need to supply the data file for the mutation prediction.
                        To create a data file, create a CSV file consisting of only 4 columns: Feature 1, Feature 2, Feature 3,
                        Feature 4 and Target</p>
                    <p>Upload the file in the webserver by clicking on "Upload" button. After uploading the file, click on
                        submit and it will start the model prediction on your data. After the prediction is done, you will see a
                        prediction graph with residue position on X-axis and Mutational Probabilityon Y-axis. You can download the
                        graph and the data in a csv format.</p>
                </div>
            </div>
            <!-- End of New Users Section -->
        </main>


        <!-- End of Main Content -->

        <!-- Right Section -->
        <div class="right-section">
            <div class="nav">
                <button id="menu-btn">
                    <span class="material-icons-sharp">
                        menu
                    </span>
                </button>
                <div class="dark-mode">
                    <span class="material-icons-sharp">light_mode</span>
                    <span class="material-icons-sharp active">dark_mode</span>
                </div>

                <div class="profile">
                    <div class="info">
                        <p id="user-greeting">Hey, <b id="user-name-placeholder">{{ user.username}}</b></p>
                    </div>
                    <div class="profile-photo">
                        <img src="images/profile-1.jpg">
                    </div>
                </div>

            </div>
            <!-- End of Nav -->

        </div>


    </div>

    <script src="../static/js/orders.js"></script>
    <script src="../static/js/dashboard.js"></script>
</body>

</html>
